src/ocl_utils_mod.f90

```diff
@@ -46,16 +57,23 @@ subroutine init_device(device, version_str, context)
                             num_platforms)
     call check_status('clGetPlatformIDs', ierr)
 
-    ! Get device IDs only for platform 1
-    iplatform=1
+    ! Select the OpenCL platform to use
+    CALL get_environment_variable("FORTCL_PLATFORM", strvalue, status=ierr)
+    if(ierr .eq. 1) then
+        ! By default use platform 1
+        iplatform=1
+    else
+        read(strvalue,"(i1)") iplatform
```

Comment:
We should probably do a bit of error checking here since we're dealing with a form of user input. I think you can get an error code back and thus check that the conversion to an integer was successful.

Reply:
I added a couple of checks regarding after this statement.